src/propgrid/property.cpp

```diff
@@ -297,13 +322,34 @@ bool wxPGDefaultRenderer::Render( wxDC& dc, const wxRect& rect,
                 imageOffset += wxCC_CUSTOM_IMAGE_MARGIN2 + 4;
             }
 
-            DrawCaptionSelectionRect( const_cast<wxPropertyGrid*>(propertyGrid), dc,
-                                      rect.x+wxPG_XBEFORETEXT-wxPG_CAPRECTXMARGIN+imageOffset,
+            const wxRect rectCaption( rect.x+wxPG_XBEFORETEXT-wxPG_CAPRECTXMARGIN+imageOffset,
                                       rect.y-wxPG_CAPRECTYMARGIN+1,
                                       ((wxPropertyCategory*)property)->GetTextExtent(propertyGrid,
                                                                                      propertyGrid->GetCaptionFont())
                                       +(wxPG_CAPRECTXMARGIN*2),
                                       propertyGrid->GetFontHeight()+(wxPG_CAPRECTYMARGIN*2) );
+
+#if WXWIN_COMPATIBILITY_3_0
+            wxPGCellRendererDrawCaptionSelectionRectFlag = true;
+            DrawCaptionSelectionRect( dc,
+                                      rectCaption.x, rectCaption.y,
+                                      rectCaption.width, rectCaption.height );
+            if ( wxPGCellRendererDrawCaptionSelectionRectFlag )
+            {
+                // This means that the user-defined overridden version of the
+                // function was called -- because the base class one was not.
+                // So just reset the flag and don't do anything else.
+                wxPGCellRendererDrawCaptionSelectionRectFlag = false;
+            }
+            else
+                // This means that our own version was called, so we now need
+                // to call the other overload, to use the user-defined version
+                // if any or to just draw the rectangle in the base class
+                // version otherwise.
+#endif // WXWIN_COMPATIBILITY_3_0
+            DrawCaptionSelectionRect( const_cast<wxPropertyGrid*>(propertyGrid), dc,
+                                      rectCaption.x, rectCaption.y,
+                                      rectCaption.width, rectCaption.height );
```

Comment:
To me the intent of this this logic is not clear but if you want to do something like this, the better place would be probably  `wxPGDrawFocusRect`, where you know what `DrawCaptionSelectionRect` calls it (win ==/ != nullptr) and where you already have conditional blocks for v3.0 compatibility code.

Reply:
The intent here is to call the overridden function with the old signature as this is what's really needed for compatibility sand this wasn't done any longer after the latest changes (before them the old function was called but the new one was not, which wasn't ideal neither).

This couldn't be done solely in `wxPGDrawFocusRect()`.

Reply:
OK, but I still don't get why do you insist on using `DrawCaptionSelectionRect()` with old signature in wx code in v3.0 mode.
In this mode `DrawCaptionSelectionRect()` is defined in two versions: with old and new signature, so why not to use only new one internally? I think version with old signature should be only used (potentially) by user app (in v3.0 mode).

Reply:
It's a virtual function, so I assume that it can be overridden in the user code, so we need to call it. If not, why was it virtual in the first place?

Reply:
OK, good point. But maybe simpler implementation would be just to call in v3.0 mode `DrawCaptionSelectionRect()` with old signature from `DrawCaptionSelectionRect()` with new signature? I mean something like this:
```
DrawCaptionSelectionRect(wxWindow*, wxDC& dc,...)
{
#if v3.0
DrawCaptionSelectionRect(dc, ...)
#else
wxPGDrawFocusRect(...)
#endif
}
```

Reply:
This means enabling 3.0 compatibility (which might be needed for something completely unrelated) would make wxPG draw uglier looking focus rect by default, which doesn't seem good. I.e. the goal is to always call `wxPGDrawFocusRect()` with a valid window by default, but still call overridden virtual function with 3.0-compatible signature if 3.0 compatibility is enabled.

And I still don't see any simpler/better way to do it than the hack in this PR.

Reply:
@a-wi Do you still object to merging this in the current state and/or see a better way to handle this?

Reply:
I'm going to push this, we can always change it later if you find a better way to do it.